api/src/main/java/org/apache/iceberg/transforms/Hours.java

```diff
@@ -33,16 +33,19 @@ static <T> Hours<T> get() {
   @Override
   @SuppressWarnings("unchecked")
   protected Transform<T, Integer> toEnum(Type type) {
-    if (type.typeId() == Type.TypeID.TIMESTAMP) {
-      return (Transform<T, Integer>) Timestamps.HOUR;
+    switch (type.typeId()) {
+      case TIMESTAMP:
+        return (Transform<T, Integer>) Timestamps.HOUR_FROM_MICROS;
+      case TIMESTAMP_NANO:
+        return (Transform<T, Integer>) Timestamps.HOUR_FROM_NANOS;
+      default:
+        throw new IllegalArgumentException("Unsupported type: " + type);
```

Comment:
nit: I think it would be good to have a unit test that exercises this path

Reply:
done